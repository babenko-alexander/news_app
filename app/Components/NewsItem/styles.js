import EStyleSheet from "react-native-extended-stylesheet";

const styles = EStyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "row",
    marginBottom: 5,
    backgroundColor: "$primaryBackgroundColor",
    height: 145,
    paddingTop: 5 
  },
  secondaryText: {
    padding: 5,
    color: "$secondaryColor",
    borderColor: "$primaryBorderColor",
    fontSize: "$secondaryFontSize",
    textAlign: "justify"
  },
  textBox: {
    flex: 1,
    justifyContent: "space-between",
  },
  greyBG: {
    backgroundColor: 'rgba(0, 0, 0, 0.4)'
  }
});

export default styles;
