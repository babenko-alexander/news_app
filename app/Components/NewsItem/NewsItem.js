import React, { PureComponent } from "react";
import { View, Text, Image, TouchableOpacity } from "react-native";
import styles from "./styles";

export default class NewsItem extends PureComponent {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <TouchableOpacity style={ this.props.fresh ? [styles.container, styles.greyBG] : styles.container} onPress={this.props.activateArticle}>
        <Image
          style={{ width: 135, height: 135 }}
          source={{
            uri: this.props.imgUrl
          }}
        />
        <View style={styles.textBox}>
          <Text style={styles.secondaryText}> {this.props.title} </Text>
          <Text style={styles.secondaryText}>
            {this.props.date}
          </Text>
        </View>
      </TouchableOpacity>
    );
  }
}
