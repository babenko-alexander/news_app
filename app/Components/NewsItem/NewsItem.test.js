import React from 'react';
import {shallow} from 'enzyme';
import NewsItem from './NewsItem';

describe('NewsItem', () => {
        it('should match to snapshot - Primary', () => {
            const component = shallow(<NewsItem text="test text"/>)
            expect(component).toMatchSnapshot("Primary NewsItem snapshot")
        });
        it('should match to snapshot - Secondary', () => {
            const component = shallow(<NewsItem text="test text" fresh={true} />)
            expect(component).toMatchSnapshot("Secondary NewsItem snapshot")
      });
});