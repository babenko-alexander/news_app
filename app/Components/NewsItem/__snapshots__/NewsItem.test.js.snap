// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsItem should match to snapshot - Primary: Primary NewsItem snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewsItem
    text="test text"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "activeOpacity": 0.2,
      "children": Array [
        <Image
          source={
            Object {
              "uri": undefined,
            }
          }
          style={
            Object {
              "height": 135,
              "width": 135,
            }
          }
        />,
        <View>
          <Text>
             
             
          </Text>
          <Text />
        </View>,
      ],
      "onPress": undefined,
      "style": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "source": Object {
            "uri": undefined,
          },
          "style": Object {
            "height": 135,
            "width": 135,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Text>
               
               
            </Text>,
            <Text />,
          ],
          "style": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                " ",
                undefined,
                " ",
              ],
              "style": undefined,
            },
            "ref": null,
            "rendered": Array [
              " ",
              undefined,
              " ",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": undefined,
              "style": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeOpacity": 0.2,
        "children": Array [
          <Image
            source={
              Object {
                "uri": undefined,
              }
            }
            style={
              Object {
                "height": 135,
                "width": 135,
              }
            }
          />,
          <View>
            <Text>
               
               
            </Text>
            <Text />
          </View>,
        ],
        "onPress": undefined,
        "style": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "source": Object {
              "uri": undefined,
            },
            "style": Object {
              "height": 135,
              "width": 135,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Text>
                 
                 
              </Text>,
              <Text />,
            ],
            "style": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  " ",
                  undefined,
                  " ",
                ],
                "style": undefined,
              },
              "ref": null,
              "rendered": Array [
                " ",
                undefined,
                " ",
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": undefined,
                "style": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`NewsItem should match to snapshot - Secondary: Secondary NewsItem snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewsItem
    fresh={true}
    text="test text"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "activeOpacity": 0.2,
      "children": Array [
        <Image
          source={
            Object {
              "uri": undefined,
            }
          }
          style={
            Object {
              "height": 135,
              "width": 135,
            }
          }
        />,
        <View>
          <Text>
             
             
          </Text>
          <Text />
        </View>,
      ],
      "onPress": undefined,
      "style": Array [
        undefined,
        undefined,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "source": Object {
            "uri": undefined,
          },
          "style": Object {
            "height": 135,
            "width": 135,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Text>
               
               
            </Text>,
            <Text />,
          ],
          "style": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                " ",
                undefined,
                " ",
              ],
              "style": undefined,
            },
            "ref": null,
            "rendered": Array [
              " ",
              undefined,
              " ",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": undefined,
              "style": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeOpacity": 0.2,
        "children": Array [
          <Image
            source={
              Object {
                "uri": undefined,
              }
            }
            style={
              Object {
                "height": 135,
                "width": 135,
              }
            }
          />,
          <View>
            <Text>
               
               
            </Text>
            <Text />
          </View>,
        ],
        "onPress": undefined,
        "style": Array [
          undefined,
          undefined,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "source": Object {
              "uri": undefined,
            },
            "style": Object {
              "height": 135,
              "width": 135,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Text>
                 
                 
              </Text>,
              <Text />,
            ],
            "style": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  " ",
                  undefined,
                  " ",
                ],
                "style": undefined,
              },
              "ref": null,
              "rendered": Array [
                " ",
                undefined,
                " ",
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": undefined,
                "style": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
